ecr/internal/testdata/digest.go

```diff
@@ -11,3 +18,12 @@ const (
 	// ImageDigest is used for image digests in tests.
 	ImageDigest = InsignificantDigest
 )
+
+// GenerateDigest returns a psuedo-random digest for use as a distinct digest in
+// tests.
+func GenerateDigest() digest.Digest {
+	d := digest.SHA256.Digester()
+	fmt.Fprintf(d.Hash(), "%s", time.Now().UTC().Format(time.RFC3339Nano))
+	fmt.Fprintf(d.Hash(), "%d", rand.Int()) // doesn't need to be very random, just sort of.
```

Comment:
Is `rand` seeded somewhere?

Reply:
To be honest, I consciously decided to not call seed at the time I wrote this. It's not cryptographically concerned and needs only be pseudo-random to produce a "different" hash value on each call.

Happy to add a call to seed if you'd like.

Reply:
Without a call to Seed, you'll get roughly deterministic responses (it acts like `rand.Seed(1)` was called), rather than pseudo-random.  I think if you want non-determinism, you need to call Seed.

Reply:
Yeah, the goal isn't non-determinism, the goal is to supply *some* generated value instead of a static hardcoded value.

If you're okay with my ack of that, I can drop a comment here explaining why.

Reply:
I'm okay with determinism in tests, but would like it to be explicit.  I think that you'll want to create a separate instance of a `rand.Rand` with an explicit constant seed.